src/AssociatedJobInfoClient.cs

```diff
@@ -0,0 +1,124 @@
+        }
+    }
+
+    public class AssociatedJobInfoClient
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly ILogger<AssociatedJobInfoClient> _logger;
+
+        public AssociatedJobInfoClient(
+            IHttpClientFactory httpClientFactory,
+            ILogger<AssociatedJobInfoClient> logger)
+        {
+            _httpClientFactory = httpClientFactory;
+            _logger = logger;
+        }
+
+        public async Task<AssociatedJobInfo> TryGetAssociatedJobInfo(
+            string getAssociatedJobUrl,
+            string authenticationToken)
+        {
+            try
+            {
+                var agentRequestJob = await TryGetAssociatedAgentRequestJob(
+                    getAssociatedJobUrl,
+                    authenticationToken);
+
+                if (agentRequestJob != null)
+                {
+                    return ParseAssociatedJobInfo(agentRequestJob);
+                }
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(
+                    exception,
+                    "Unable to get associated job info from {getAssociatedJobUrl} because of an exception",
+                    getAssociatedJobUrl);
+            }
+
+            return AssociatedJobInfo.Empty;
+        }
+
+        private async Task<AgentRequestJob> TryGetAssociatedAgentRequestJob(
+            string getAssociatedJobUrl,
+            string authenticationToken)
+        {
+            if (!string.IsNullOrEmpty(getAssociatedJobUrl))
+            {
+                _logger.LogInformation("Getting associated job from {AssociatedJobUrl}", getAssociatedJobUrl);
+
+                var httpClient = _httpClientFactory.CreateClient();
+                var message = new HttpRequestMessage(HttpMethod.Get, getAssociatedJobUrl);
+                message.Headers.Authorization = new AuthenticationHeaderValue("Bearer", authenticationToken);
+                var response = await httpClient.SendAsync(message);
+                if (response.IsSuccessStatusCode)
+                {
+                    var responseJson = await response.Content.ReadAsStringAsync();
+                    var responseData = JsonConvert.DeserializeObject<AgentRequestJob>(responseJson);
+                    return responseData;
+                }
+                else
+                {
+                    _logger.LogInformation("GetAssociatedJob responded with error code {StatusCode}", response.StatusCode);
+                }
+            }
+            else
+            {
+                _logger.LogInformation("Unable to get associated job info because getAssociatedJobUrl is not set");
+            }
+
+            return null;
+        }
+
+        private AssociatedJobInfo ParseAssociatedJobInfo(AgentRequestJob job)
+        {
+            var buildSourceBranch = GetVarValueOrEmpty("build.sourceBranch", job.Job.Variables);
```

Comment:
There are a lot of "var" in this code, which goes against the coding style guidelines.

Reply:
Ok, I can change it. To make sure - do we have a guideline to avoid `var` or do we use the .NET ones, i.e. use `var` unless the type is not clear?

Reply:
We just use the existing style guidelines. https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md, rule 10.  Only "new Thing" and a hard cast of "Thing" are places where "var" is ok (because the type name is directly on the line anyway).

Reply:
I've changed it